lib/skate/detours/detours.ex

```diff
@@ -141,6 +122,34 @@ defmodule Skate.Detours.Detours do
   """
   def get_detour!(id), do: Repo.get!(Detour, id)
 
+  @doc """
+  Gets a single detour with state intact.
+
+  Raises `Ecto.NoResultsError` if the Detour does not exist.
+
+  ## Examples
+
+      iex> get_detour_with_state!(123)
+      %DetourWithState{}
+
+      iex> get_detour_with_state!(456)
+      ** (Ecto.NoResultsError)
+
+  """
+  @spec get_detour_with_state!(integer()) :: DetourWithState.t()
+  def get_detour_with_state!(id) do
+    detour =
+      Detour
+      |> Repo.get!(id)
+      |> Repo.preload(:author)
```

Comment:
suggestion: I would either replace the `Repo.get!` with `get_detour!`, or do the preload in a single query

1. 
```suggestion
      get_detour!(id)
      |> Repo.preload(:author)
```

```suggestion

      Detour
      from d in Detour, where: d.id == ^id, preload: [:author]
      |> Repo.one!()
```